R/data-converters.R

```diff
@@ -65,19 +65,33 @@ enw_add_cumulative <- function(obs, by = NULL, copy = TRUE) {
 #' # Make use of maximum reported to calculate empirical daily reporting
 #' dt <- enw_add_max_reported(dt)
 #' enw_add_incidence(dt)
+
+# --------------
+##  temporary placeholder
+enw_check_date_sequence  <- function(obs, by=NULL, copy=TRUE) {
+
+  obs <- obs[, .SD[reference_date >= min(report_date) | is.na(reference_date)],
+    by = by
+  ]
+}
+# --------------
+
 enw_add_incidence <- function(obs, set_negatives_to_zero = TRUE, by = NULL,
                               copy = TRUE) {
   reports <- coerce_dt(
     obs, dates = TRUE, required_cols = c(by, "confirm"), copy = copy
   )
   data.table::setkeyv(reports, c(by, "reference_date", "report_date"))
+
   reports[, new_confirm := c(confirm[1], diff(confirm)),
     by = c("reference_date", by)
   ]
-  reports <- reports[,
-    .SD[reference_date >= min(report_date) | is.na(reference_date)],
-    by = by
-  ]
+
+# --------------
+  ## temporary
+  reports  <- enw_check_date_sequence(reports)
```

Comment:
I could be completely misunderstanding (@adrian-lison) but my understanding of #305 was that it would be good to move this filter out of enw_add_incidence so that you would call this and then then call `enw_add_incidence`?  I am not totally sure though?